src/utils/detect-server-settings.mjs

```diff
@@ -250,9 +253,14 @@ const detectChangesInNewSettings = (frameworkSettings, newSettings, metadata) =>
   }
 }
 
-const detectFrameworkSettings = async ({ projectDir }) => {
-  const projectFrameworks = await listFrameworks({ projectDir })
-  const frameworks = projectFrameworks.filter((framework) => hasDevCommand(framework))
+/**
+ * @param {Project} project
+ */
+const detectFrameworkSettings = async (project) => {
+  const projectFrameworks = (await project.detectFrameworks()) ?? []
```

Comment:
To have the correct commands for build systems we need to call `getBuildSettings` as outlined in the issues.

otherwise, we would not get the right command for build tools like nx or turbo

Reply:
i'm not super confident around this part of the codebase, but I *think* that https://github.com/netlify/cli/pull/5816/commits/10669e3d94637e1a2b89026b0e71ce627e961a86 implements what you're referring to 🤞